### PR TITLE
[NEW-UI]  Handle delete task action edge-cases

### PR DESCRIPTION
**Pull request by zankevich**

### CODE REVIEW

**Comment by bsekachev** [comment]
@zankevich 

I put breakpoints in the lambda returned by deleteTaskAsync().
There are several issues. 
* First of all, taskFilter.currentPage and taskFilter.searchQuery are undefined for me (probably it's because I don't search anything and I have only one page). But there aren't any checks in the code and we can get queryObject.state === NaN (undefined - 1 = NaN)
* The variable state.tasks.tasks.length isn't changed for me. I remove tasks (and they are actually removed), but this field stays const. 

Other issue is that nothing happens with task blocks after them are removed. So, in order to task disappears from the page, I need update a page. 

Also we noticed, that immediately after loading page (dashboard or login), server gets login request with invalid credentials. Probably there is extra line somewhere.

**Comment by zankevich** [comment]
> @zankevich
> 
> I put breakpoints in the lambda returned by deleteTaskAsync().
> There are several issues.
> 
> * First of all, taskFilter.currentPage and taskFilter.searchQuery are undefined for me (probably it's because I don't search anything and I have only one page). But there aren't any checks in the code and we can get queryObject.state === NaN (undefined - 1 = NaN)
> * The variable state.tasks.tasks.length isn't changed for me. I remove tasks (and they are actually removed), but this field stays const.
> 
> Other issue is that nothing happens with task blocks after them are removed. So, in order to task disappears from the page, I need update a page.
> 
> Also we noticed, that immediately after loading page (dashboard or login), server gets login request with invalid credentials. Probably there is extra line somewhere.

Hey, good questions!

> First of all, taskFilter.currentPage and taskFilter.searchQuery are undefined for me (probably it's because I don't search anything and I have only one page). But there aren't any checks in the code and we can get queryObject.state === NaN (undefined - 1 = NaN)

I suppose you mean queryObject.**page** === NaN (undefined - 1 = NaN).
This is fixed. Now I added extra check for if it's a last task.

> The variable state.tasks.tasks.length isn't changed for me. I remove tasks (and they are actually removed), but this field stays const.

This is because the state is not yet updated in the App. This is not a problem because It's what we need (we need to check if task was latest on the page).

> Other issue is that nothing happens with task blocks after them are removed. So, in order to task disappears from the page, I need update a page.

It was a bug and now it is fixed.

> Also we noticed, that immediately after loading page (dashboard or login), server gets login request with invalid credentials. Probably there is extra line somewhere.

Can you add some steps to reproduce?

BTW: [here](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd) you can find a tool to check app state. It can be helpful to see whats going up in the app.

File: cvat-ui/src/actions/auth.actions.ts [hunk 1]
```diff
@@ -17,13 +17,15 @@ export const loginError = (error = {}) => (dispatch: any) => {
...
 }
 
-export const loginAsync = (username: string, password: string) => {
+export const loginAsync = (username: string, password: string, history: any) => {
   return (dispatch: any) => {
     dispatch(login());
 
     return (window as any).cvat.server.login(username, password).then(
       (authenticated: any) => {
+        localStorage.setItem('session', 'true');
```
Comment (bsekachev) [inline cvat-ui/src/actions/auth.actions.ts]: We use localStorage to define whether we would be redirected? I am not sure it's a good idea.

File: cvat-ui/src/components/page-not-found/page-not-found.test.tsx [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import NotFound from './not-found';
+
+import NotFound from './page-not-found';
```
Comment (bsekachev) [inline cvat-ui/src/components/page-not-found/page-not-found.test.tsx]: PageNotFound instead of NotFound?

**Comment by bsekachev** [comment]
@zankevich 
Previous issues have been resolved. 

In this code I see too many "any" types. As far as I understand it isn't good for typescript project. Can we follow more strict typification, or we don't have any choice?

File: cvat-ui/src/components/page-not-found/page-not-found.test.tsx (hunk 2 above)
Comment (zankevich) [reply cvat-ui/src/components/page-not-found/page-not-found.test.tsx]: I added `page` for all pages in app.

**Comment by zankevich** [comment]
> @zankevich
> Previous issues have been resolved.
> 
> In this code I see too many "any" types. As far as I understand it isn't good for typescript project. Can we follow more strict typification, or we don't have any choice?

Sometimes it is not so easy to find a correct type (especially when using libs). I'll plan to add more types later.

**Pull request merged**